src/core/Akka.Remote/Transport/TestTransport.cs

```diff
@@ -17,44 +17,42 @@
 namespace Akka.Remote.Transport
 {
     /// <summary>
-    /// Transport implementation used for testing.
-    /// 
-    /// The TestTransport is basically shared memory between actor systems. It can be programmed to emulate
-    /// different failure modes of a <see cref="Transport"/> implementation. TestTransport keeps a log of the activities
-    /// it was requested to do. This class is not optimized for performance and MUST not be used in production systems.
+    ///     Transport implementation used for testing.
+    ///     The TestTransport is basically shared memory between actor systems. It can be programmed to emulate
+    ///     different failure modes of a <see cref="Transport" /> implementation. TestTransport keeps a log of the activities
+    ///     it was requested to do. This class is not optimized for performance and MUST not be used in production systems.
     /// </summary>
     public class TestTransport : Transport
     {
-
-
-        public readonly Address LocalAddress;
+        private readonly TaskCompletionSource<IAssociationEventListener> _associationListenerPromise =
+            new TaskCompletionSource<IAssociationEventListener>();
 
         private readonly AssociationRegistry _registry;
+        public readonly SwitchableLoggedBehavior<Address, AssociationHandle> AssociateBehavior;
+        public readonly SwitchableLoggedBehavior<TestAssociationHandle, bool> DisassociateBehavior;
+        /*
+         * Programmable behaviors
+         */
 
-        private TaskCompletionSource<IAssociationEventListener> _associationListenerPromise
```

Comment:
This was a bug from earlier - returned a new `TaskCompletionSource<IAssociationEventListener>` each time it was called.